src/Agent.Plugins/Artifact/FileContainerProvider.cs

```diff
@@ -51,7 +51,9 @@ public async Task DownloadMultipleArtifactsAsync(ArtifactDownloadParameters down
         {
             foreach (var buildArtifact in buildArtifacts)
             {
-                var dirPath = Path.Combine(downloadParameters.TargetDirectory, buildArtifact.Name);
+                var dirPath = downloadParameters.AppendArtifactNameToTargetPath
```

Comment:
If we download multiple artifacts in DownloadBuildArtifactsV0, does it create folders?

Reply:
What happens in both cases if there are two `file.txt` in two different artifacts?

Reply:
In [`ResolveTargetPath`](https://github.com/microsoft/azure-pipelines-agent/blob/master/src/Agent.Plugins/Artifact/FileContainerProvider.cs#L285) function we append `FileContainerItem.Path` to the `rootPath`, which is the same as `dirPath` here. `FileContainerItem.Path` already includes the artifact name, so appending `buildArtifact.Name` in the line you've highlighted produces paths like this:
`$(System.ArtifactsDirectory)/drop1/drop1/file.txt`. In this example, `$(System.ArtifactsDirectory)/drop1` is the `dirPath`, and `drop1/file.txt` is `FileContainerItem.Path`

Reply:
So, if there are two artifacts, drop1 and drop2, the paths before the changes would be:
`$(System.ArtifactsDirectory)/drop1/drop1/file.txt` and
`$(System.ArtifactsDirectory)/drop2/drop2/file.txt`

After the changes:
`$(System.ArtifactsDirectory)/drop1/file.txt` and
`$(System.ArtifactsDirectory)/drop2/file.txt`

So, no conflicts should arise

Reply:
Awesome, thanks!